aiida/storage/psql_dos/orm/nodes.py

```diff
@@ -182,55 +183,60 @@ def user(self, user):
         type_check(user, self.USER_CLASS)
         self.model.user = user.bare_model
 
-    def add_incoming(self, source, link_type, link_label):
+    @contextmanager
+    def get_session(self):
```

Comment:
As above for groups, this should be on the backend, rather the duplicating it here, and possibly just use the existing `transaction` context manager